### DISCUSSION TITLE
native port disconnected / incomplete file downloads

### DISCUSSION THREAD

**Discussion by ghost** [aclap-dev/video-downloadhelper#252] [Help and Q&A]
### Your question

Good afternoon

I have been having issues with my downloads either getting the error "native port disconnected" or the file only partially downloads and completes with no errors (still only downloading part of the video).  The size of the file does not matter either, nor do the websites I have tried using it on.

I have read about the DMR in firefox and I have tried that with it on and off and I get the same results.  I have tried re installing both firefox, the addon and the companion app with the same results.  I wasnt sure what else to search before I posted my own issue.  I apologize if it is a redundant post.

I am at an all time loss as I have never had issues with the addon before.  This is also a new laptop (new to me).  But like I said I have had this addon and firefox on at least 6 different systems and this is the first time I am getting these errors.

Any help would be greatly appreciated.  have a great day!

### Addon & CoApp version and details

version: 8.2.1.1
channel: stable
build date: 2024-02-19
build options: linuxlic: false, noyt: false, browser: firefox
lang: en-US
license: unset
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:123.0) Gecko/20100101 Firefox/123.0
coapp: yes. Info:
{
  "id": "net.downloadhelper.coapp",
  "name": "VdhCoApp",
  "version": "2.0.19",
  "binary": "C:\\Program Files\\DownloadHelper CoApp\\vdhcoapp.exe",
  "displayName": "VdhCoApp",
  "description": "Video DownloadHelper companion app",
  "target": {
    "os": "windows",
    "arch": "x86_64",
    "node": "18"
  },
  "home": "C:\\Users\\Master",
  "converterBinary": "C:\\Program Files\\DownloadHelper CoApp\\ffmpeg.exe",
  "converterBase": "ffmpeg",
  "converterBaseVersion": "c30f360955"
}

**Comment by paulrouget** [comment]
Can you please share an URL to a failing video?

**Comment by ghost** [comment]
It is literally any site for any video.  Specifically I download from pornhub and spankbang. (no shame here)
Fresh install of windows as well so no addons or settings changed or anything like that.

**Reply by GrampaWildWilly** [reply]
I am running Windows 11 Pro 64-bit version 23H2 build 22631.3235, Firefox 123.0 64-bit, licensed VDH 8.2.2.3a1 beta, CoApp 2.0.19.  I forgot to check before I ran this test.  I have since updated to VDH 8.2.2.4a1 beta, which apparently came out within the past few hours while I wasn't looking.  I'm sure it would have worked equally well for this test.  Let me point out that I do not work for VDH.  I am just another user.

If it's literally any site then you need to literally supply literally 2 or 3 URLs that are literally giving you literally problems.  You need to literally supply enough information for literally somebody else to try to literally replicate your results.  Literally always.

Every so often somebody comes in here & claims they can't download things from Pornhub.  Going back to VDH 6 I have had only intermittent trouble downloading from Pornhub.  I say intermittent because every once in a while, something won't download.  But a retry a few minutes or hours later works fine.  I believe this is more about the site than anything I've been doing.  In my test just now, everything ran fine.  That was:

https://www.pornhub.com/view_video.php?viewkey=64d18da621084

When I first visited the page, without even clicking the play button in the player, I got this:

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/ac81d0b2-97dd-4388-ad10-8d10d5e76709)
![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/552ee537-9fec-477c-ac26-f509ce011412)

Two things to notice there.  First, the menu was long so it took 2 images to show it all.  Second, I have heavily censored them.  This should be the standard practice for anybody dealing with similar content.

In any case, I downloaded the first variant on the VDH menu shown in the first image.  I changed the name of the target file, as you'll notice in a moment.  Here it is downloading.

![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/9977994c-8028-4662-b64a-3ad08a74e7af)

I had to be quick because it didn't take very long to complete.

![#04](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/b6b55380-7ac3-41b3-bc49-ca807f6b2d1c)

It played fine in VLC with both video & audio all the way through.  Since this features two of my favorites, Aidra Fox & the late lamented Dahlia Sky, I did sit & watch it.  I would show you an image from VLC but . . . you know.

So my usual advice applies.  Get on the latest releases of everything.  That means the VDH beta.  You can find that -- and many other useful things -- by clicking the Wiki link at the top of this page.  The Wiki page shows a Table of Contents down the right side.  You would do well to look at all of the chapters, but for the moment, just go to the chapter on Beta versions.  You should be able to take it from there.

**Reply by ghost** [reply]
As i appreciate your answer, I stopped reading after your  condescending tone with the use of the word literal.  I apologize if we are not all as smart as you.  I would rather go find the answer somewhere else than spoken to that way.  Have a great day.

**Reply by GrampaWildWilly** [reply]
I just meant it as humor.  I meant no condescension.

**Comment by paulrouget** [comment]
> It is literally any site for any video. Specifically I download from pornhub and spankbang. (no shame here).

Can you please still share one failing URL still? So I'm sure I'm trying to reproduce with the exact same setup.

**Reply by lemelomane** [reply]
I have the same issue when trying to download in a protected Onlyfans video with DRM protection.

**Reply by paulrouget** [reply]
https://github.com/aclap-dev/video-downloadhelper/wiki/What-can-I-do-if-the-video-is-protected-by-DRM%3F

**Reply by GrampaWildWilly** [reply]
That is a great chapter in the Wiki.  But I would suggest you rearrange the title so that the acronym DRM appears in the part of the title that is displayed by default on the Wiki page.  Perhaps something like this:

What if DRM is protecting my video?

The important content of the title is DRM but at the moment the occurrence of DRM is hidden by an ellipsis.  You need to change the title so DRM is visible on the Wiki page.
